websockify/websocketproxy.py

```diff
@@ -245,13 +257,19 @@ def do_proxy(self, target):
                 # Receive target data, encode it and queue for client
                 buf = target.recv(self.buffer_size)
                 if len(buf) == 0:
+
+                    # Send queued target data to the client
+                    c_pend = self.send_frames(cqueue)
+                    cqueue = []
```

Comment:
This does nothing as far as I can tell. If there was any data in `cqueue`, then `send_frames()` would have already been called.

The buffering issue is deeper, where you need to keep calling `send_frames()` until `c_pend` is `False`.

I'm concerned what tests you were doing, though. I would have assumed you would have seen a lack of flush here?